affiliation/identity.go

```diff
@@ -0,0 +1,437 @@
+		if err != nil {
+			log.Println("GetOrganizations: Could not get the organizations: ", err)
+			return nil
+		}
+
+		var errMsg AffiliationsResponse
+		err = json.Unmarshal(res, &errMsg)
+		if err != nil || errMsg.Message != "" {
+			log.Println("GetOrganizations: failed to get organizations: ", errMsg)
+		}
+
+		time.Sleep(2 * time.Minute)
+		return a.GetOrganizations(uuid, projectSlug)
+	}
+
+	var response EnrollmentsResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		log.Println("GetOrganizations: failed to unmarshal enrollments response: ", err)
+		return nil
+	}
+	return &response.Enrollments
+}
+
+// GetProfile ...
+func (a *Affiliation) GetProfile(uuid, projectSlug string) *ProfileResponse {
+	if uuid == "" || projectSlug == "" {
+		return nil
+	}
+	token, err := a.auth0ClientProvider.GetToken()
+	if err != nil {
+		log.Println(err)
+	}
+	headers := make(map[string]string, 0)
+	headers["Authorization"] = fmt.Sprintf("%s %s", "Bearer", token)
+
+	endpoint := a.AffBaseURL + "/affiliation/" + url.PathEscape(projectSlug) + "/get_profile/" + uuid
+
+	_, res, err := a.httpClientProvider.Request(strings.TrimSpace(endpoint), "GET", headers, nil, nil)
+	if err != nil {
+		log.Println("GetProfile: Could not get the profile: ", err)
+		return nil
+	}
+
+	var response ProfileResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		log.Println("GetProfile: failed to unmarshal profile response: ", err)
+		return nil
+	}
+	return &response
+}
+
+// GetIdentityByUser ...
+func (a *Affiliation) GetIdentityByUser(key string, value string) (*AffIdentity, error) {
+	if key == "" || value == "" {
+		nilKeyOrValueErr := "GetIdentityByUser: key or value is null"
+		return nil, fmt.Errorf(nilKeyOrValueErr)
+	}
+	token, err := a.auth0ClientProvider.GetToken()
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	headers := make(map[string]string, 0)
+	headers["Authorization"] = fmt.Sprintf("%s %s", "Bearer", token)
+	endpoint := a.AffBaseURL + "/affiliation/" + "identity/" + key + "/" + value
+	statusCode, res, err := a.httpClientProvider.Request(strings.TrimSpace(endpoint), "GET", headers, nil, nil)
+	switch statusCode {
+	case http.StatusBadRequest, http.StatusNotFound:
+		log.Println("GetIdentityByUser: Could not get the identity: l", err)
+		return nil, errors.New("identity not found")
+	case http.StatusOK:
+	default:
+		if err != nil {
+			log.Println("GetIdentityByUser: Could not get the identity: ", err)
+			return nil, err
+		}
+
+		var errMsg AffiliationsResponse
+		err = json.Unmarshal(res, &errMsg)
+		if err != nil || errMsg.Message != "" {
+			log.Println("GetIdentityByUser: failed to get identity: ", errMsg)
+		}
+
+		time.Sleep(2 * time.Minute)
+		return a.GetIdentityByUser(key, value)
+	}
+
+	var ident IdentityData
+	err = json.Unmarshal(res, &ident)
+	if err != nil {
+		return nil, err
+	}
+
+	profileEndpoint := a.AffBaseURL + "/affiliation/" + url.PathEscape(a.ProjectSlug) + "/get_profile/" + *ident.UUID
```

Comment:
V2 isn't supposed to call affiliations API IMHO.

Reply:
Yes, you are right, it's just temp will be removed